crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -2766,6 +2792,128 @@ pub(crate) mod tests {
         assert_eq!(smartrest_fields.next().unwrap(), "up");
     }
 
+    #[test_case("restart")]
+    #[test_case("software_list")]
+    #[test_case("software_update")]
+    #[test_case("log_upload")]
+    #[test_case("config_snapshot")]
+    #[test_case("config_update")]
+    #[test_case("custom_op")]
+    #[tokio::test]
+    async fn operations_not_supported_for_nested_child_devices(op_type: &str) {
+        let tmp_dir = TempTedgeDir::new();
+        let (mut converter, _http_proxy) = create_c8y_converter(&tmp_dir).await;
+
+        // Register immediate child device and nested child device
+        let reg_message = Message::new(
+            &Topic::new_unchecked("te/device/immediate_child//"),
+            json!({
+                "@type":"child-device",
+                "@parent":"device/main//",
+                "@id":"immediate_child"
+            })
+            .to_string(),
+        );
+        let _ = converter.convert(&reg_message).await;
+        let reg_message = Message::new(
+            &Topic::new_unchecked("te/device/nested_child//"),
+            json!({
+                "@type":"child-device",
+                "@parent":"device/immediate_child//",
+                "@id":"nested_child"
+            })
+            .to_string(),
+        );
+        let _ = converter.convert(&reg_message).await;
+
+        let capability_msg = Message::new(
+            &Topic::new_unchecked(&format!("te/device/nested_child///cmd/{op_type}")),
+            "[]",
+        );
+        let messages = converter.convert(&capability_msg).await;
+
+        assert_messages_matching(
+            &messages,
+            [(
+                "te/errors",
+                "operation for nested child devices are currently unsupported".into(),
+            )],
+        );
+    }
+
+    #[test_case("restart")]
+    #[test_case("software_list")]
+    #[test_case("software_update")]
+    #[test_case("log_upload")]
+    #[test_case("config_snapshot")]
+    #[test_case("config_update")]
+    #[test_case("custom_op")]
+    #[tokio::test]
+    async fn operations_not_supported_for_services(op_type: &str) {
+        let tmp_dir = TempTedgeDir::new();
+        let (mut converter, _http_proxy) = create_c8y_converter(&tmp_dir).await;
+
+        // Register main device service
+        let _ = converter
```

Comment:
I prefer `unwrap()` here instead of using `let _` because unwrap can cause a panic and test can fail here.

Reply:
I can't `unwrap` here because the `convert` function does not return a Result. It's the `try_convert` function that returns a result.

Reply:
Ah okay, I misunderstood what is returned here. So it just ignore the return of the converter.